Game/Source/GDKTestGyms/Private/EventTracerComponent.cpp

```diff
@@ -0,0 +1,85 @@
+#include "EventTracerComponent.h"
+
+#include "Interop/Connection/SpatialEventTracerUserInterface.h"
+#include "SpatialGDKFunctionalTests/SpatialGDK/EventTracingTests/EventTracingTest.h"
+#include "Interop/Connection/SpatialTraceEventBuilder.h"
+#include "Interop/Connection/UserSpanId.h"
+#include "Kismet/GameplayStatics.h"
+#include "Net/UnrealNetwork.h"
+
+UEventTracerComponent::UEventTracerComponent()
+{
+	SetIsReplicatedByDefault(true);
+	PrimaryComponentTick.bCanEverTick = true;
+}
+
+void UEventTracerComponent::GetLifetimeReplicatedProps(TArray<FLifetimeProperty> & OutLifetimeProps) const
+{
+	Super::GetLifetimeReplicatedProps(OutLifetimeProps);
+
+	DOREPLIFETIME(UEventTracerComponent, TestInt);
+}
+
+void UEventTracerComponent::BeginPlay()
+{
+	Super::BeginPlay();
+
+	UserSendRPCEventName = AEventTracingTest::GetUserSendRPCEventName().ToString();
+	UserProcessRPCEventName = AEventTracingTest::GetUserProcessRPCEventName().ToString();
+	UserSendComponentPropertyEventName = AEventTracingTest::GetUserSendComponentPropertyEventName().ToString();
+	UserReceiveComponentPropertyEventName = AEventTracingTest::GetUserReceiveComponentPropertyEventName().ToString();
+
+	if (OwnerHasAuthority() && bUseEventTracing)
+	{
+		GetWorld()->GetTimerManager().SetTimer(TimerHandle, this, &UEventTracerComponent::TimerFunction, 5.0f, true);
+	}
+}
+
+void UEventTracerComponent::TimerFunction()
+{
+	// Create a user trace event for sending a property update
+	FUserSpanId SpanId = USpatialEventTracerUserInterface::CreateSpanId(this);
+	USpatialEventTracerUserInterface::TraceEvent(this, SpanId, SpatialGDK::FSpatialTraceEventBuilder(UserSendComponentPropertyEventName).GetEvent());
+	USpatialEventTracerUserInterface::TraceProperty(this, this, SpanId);
+
+	TestInt++;
+
+	// Create a user trace event for sending an RPC
+	SpanId = USpatialEventTracerUserInterface::CreateSpanId(this);
+	USpatialEventTracerUserInterface::TraceEvent(this, SpanId, SpatialGDK::FSpatialTraceEventBuilder(UserSendRPCEventName).GetEvent());
+
+	FEventTracerRPCDelegate Delegate;
+	Delegate.BindUFunction(this, "RunOnClient");
```

Comment:
Is this how it's done? Don't think I have ever used this, but feels dirty.

Reply:
Maybe a lambda is better suited for c++ usage

Reply:
Binding a lambda to the delegate?

Reply:
Would game c++ code use the event tracer directly? We could extend the interface a bit to be nicer with regular code if not

Reply:
I suppose there is an option to bind it to the delegate